### PR TITLE
Update `eth-bridge-integration` for namada v0.10.1

### PR DESCRIPTION
**Pull request by james-chf**
Most relevant changes
- `common::SecretKey` in wallet no longer wrapped by `Rc` (affects some Ethereum hot/cold key code)
- there is a new `TxIndex` type to represent the index of a transaction in a block, used in some places

NB: errors like the below may occur now when running tests using `.wasm` files which were built natively on ARM64 macOS, `*.wasm` files need to be built with `make build-wasm-scripts-docker`

```
--- STDOUT:                       namada_tests native_vp::eth_bridge_pool::test_bridge_pool_vp::validate_mint_wnam_tx ---

running 1 test
test native_vp::eth_bridge_pool::test_bridge_pool_vp::validate_mint_wnam_tx ... FAILED

failures:

failures:
    native_vp::eth_bridge_pool::test_bridge_pool_vp::validate_mint_wnam_tx

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 58 filtered out; finished in 0.17s


--- STDERR:                       namada_tests native_vp::eth_bridge_pool::test_bridge_pool_vp::validate_mint_wnam_tx ---
thread 'native_vp::eth_bridge_pool::test_bridge_pool_vp::validate_mint_wnam_tx' panicked at 'Test failed.: ValidationError(ForbiddenWasmFeatures(BinaryReaderError { inner: BinaryReaderErrorInner { message: "sign extension operations support is not enabled", offset: 35520, needed_hint: None } }))', tests/src/native_vp/eth_bridge_pool.rs:113:29
```

### CODE REVIEW

**Comment by james-chf** [comment]
pls update wasm

**Comment by sug0** [comment]
> NB: errors like the below may occur now when running tests using `.wasm` files which were built natively on ARM64 macOS, `*.wasm` files need to be built with `make build-wasm-scripts-docker`

oh no. :( I hate docker, it uses so much space

**Comment by james-chf** [comment]
> > NB: errors like the below may occur now when running tests using `.wasm` files which were built natively on ARM64 macOS, `*.wasm` files need to be built with `make build-wasm-scripts-docker`
> 
> oh no. :( I hate docker, it uses so much space

Yeah I'm not sure what changed between v0.9.0 and v0.10.1 that you can't use natively built `*.wasm`s anymore on M1 (e2e tests will fail also). Ideally we can fix this to work again at some later point

File: apps/src/lib/node/ledger/shell/process_proposal.rs [hunk 1]
```diff
@@ -474,12 +474,20 @@ where
                         ),
                     }
                 } else {
+                    // If the public key corresponds to the MASP sentinel
+                    // transaction key, then the fee payer is effectively
+                    // the MASP, otherwise derive
+                    // they payer from public key.
```
Comment (sug0) [inline apps/src/lib/node/ledger/shell/process_proposal.rs:480]:
```suggestion
                    // the payer from public key.
```

**Comment by sug0** [comment]
> > > NB: errors like the below may occur now when running tests using `.wasm` files which were built natively on ARM64 macOS, `*.wasm` files need to be built with `make build-wasm-scripts-docker`
> > 
> > 
> > oh no. :( I hate docker, it uses so much space
> 
> Yeah I'm not sure what changed between v0.9.0 and v0.10.1 that you can't use natively built `*.wasm`s anymore on M1 (e2e tests will fail also). Ideally we can fix this to work again at some later point

looks like we might need to hack into wasmer again, this time with aarch64 codegen

**Review by batconjurer — APPROVED** [review]

**Review by sug0 — APPROVED** [review]

**Pull request merged**